apisix/utils/log-util.lua

```diff
@@ -267,6 +287,12 @@ function _M.check_log_schema(conf)
             return nil, "failed to validate the 'include_resp_body_expr' expression: " .. err
         end
     end
+    if conf.matches then
+        local ok, err = expr.new(conf.matches)
+        if not ok then
+            return nil, "failed to validate the 'matches' expression: " .. err
+        end
+    end
```

Comment:
No need to do this in the underlying function, you can do this in the plugin.
Not every plugin has this configuration `matches`

Reply:
@monkeyDluffy6017  
I think this judgment can be implemented in the underlying function. 
If other logger plugins need it, you only need to define `matches` in the `scheme` to have this feature

@Gallardot WDYT?

Reply:
But no other plugins need it now, we can move it here anytime if in need

Reply:
got it, i fix it